### DISCUSSION TITLE
[GitHub Actions] branch conditional

### DISCUSSION THREAD

**Discussion by troygoode** [community/community#26349] [Actions]
<p>I’d like to make a job or step (either would work for me) conditional in our workflow, and have it validate the current branch is “master”. Is there something we can reference in the <code>if:</code> conditional to match only when the branch being pushed to is master?</p>

**Comment by Geertvdc** [comment, marked as answer]
<p>You can do a conditional statement using this:</p>
<pre><code>if: github.ref == 'refs/heads/master'
</code></pre>

**Comment by cmygray** [comment]
<p>How did you know the value of <code>github.ref</code>? I can’t find it.</p>

**Comment by xmo-odoo** [comment]
<p><a href="https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener">https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions#github-context</a></p>
<p>“ref” is a term of art in git (and thus github, it’s a common encounter in the API): <a href="https://git-scm.com/book/en/v2/Git-Internals-Git-References" rel="nofollow noopener">https://git-scm.com/book/en/v2/Git-Internals-Git-References</a></p>

**Comment by pqvst** [comment]
<p>You can print the value to verify:</p>
<pre><code>steps:
    - run: echo ${{github.ref}}
</code></pre>

**Comment by tarun1475** [comment]
<p>if: github.ref == ‘refs/heads/integration’ &amp;&amp; github.event_name == ‘push’ </p>
<p>You can this command and replace whatever branch or event you want to run for.</p>

**Comment by goetzc** [comment]
<p>To archive the same with multiple branches and for less boilerplate, use the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#functions" rel="nofollow noopener"><code>contains</code></a> function:</p>
<pre><code class="lang-auto">    if:
      contains('
        refs/heads/staging
        refs/heads/production
      ', github.ref)
</code></pre>
<p>More information: <a href="https://stackoverflow.com/a/62419599/2574719" rel="nofollow noopener">https://stackoverflow.com/a/62419599/2574719</a></p>

**Comment by sindrehauglum** [comment]
<p>Include this in yout workflow.</p>
<pre><code class="lang-auto"># Controls when the action will run. Triggers the workflow on push 
# events but only for the master branch
on:
  push:
    branches: [ master ]
</code></pre>

**Comment by gempain** [comment]
<p>This approach may lead to unexpected results as it will also match when the branch name is refs/heads/prod or any substring of <code>production</code> or <code>staging</code>.</p>

**Comment by KrishnaMohan9** [comment]
I have another question. Someone suggest or tell me an idea.

We have two repositories.

1. Application Code with different branches and it consists a Yaml
2. Build and Deploy Template Yaml in other repo.

From the Application code branch, we call another repo's build template. 

My question is, from this yaml if it is a feature branch deployed in DEV and if it is the main deploy PROD? How to write a condition for the cross-repo Yaml?

Note: We cannot update the build and deploy template as it is used for other services. Any changes we should perform to our application Yaml

Application code branch yaml:
`on: 

    push:
        branches:
            - main

    workflow_dispatch:

  ```
 jobs:
      main:
        name: _ # We don't care about the name.
        uses: krishna/demo/.github/workflows/dotnet.yaml@feature/gitflow-update
        with:
            name: verification-service
            dockerfile: ./Dockerfile
      
```
        secrets:
            NUGET_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
            SONARQUBE_TOKEN : ${{ secrets.SONARQUBE_TOKEN }}`
